### PR TITLE
added all modules crud

### PR DESCRIPTION
**Pull request by ibrahimjamil**

### CODE REVIEW

File: .env.example
Comment (mateenqazi) [inline .env.example:1]: env in gitignore file. env.example is a dummy file. but it is good practice. if it is in gitignore.

File: services/project.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	projectErr, _ := ps.GetProjectByName(project.Name)
+	if projectErr == nil {
+		return errors.New("project already exists"), false
+	}
+
+	res := ps.Db.Model(&Project{}).Create(&project)
+	if res.Error != nil {
+		return errors.New("project not create successfully"), false
+	}
+	return nil, true
```
Comment (mateenqazi) [inline services/project.go:35]: return always after a line space it is good for code readability.

File: services/project.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func (ps *ProjectServiceManager) DeleteProject(id int) (error, Project) {
+	var project Project
+
+	err, fetchedProject := ps.GetProject(id)
+	if err != nil {
+		return errors.New("didnt get project"), Project{}
+	}
+
+	tx := ps.Db.Begin()
```
Comment (mateenqazi) [inline services/project.go:66]: what is the purpose of begin and commit there? it's looking good. please add a comment on it.

File: .env.example
Comment (ibrahimjamil) [reply .env.example:1]: Mateen it's for the purpose of code standard I saw this in some projects they have this so the new developer knows which env vars for this project required
so i guess its required to put in there as a sample

File: services/project.go (hunk 1 above)
Comment (ibrahimjamil) [reply services/project.go:35]: yeah done

File: services/project.go (hunk 2 above)
Comment (ibrahimjamil) [reply services/project.go:66]: it's for the purpose of executing multiple queries if one of the queries fails it didn't execute that queries in between begin and commit its generally called transactions in DB terminology.

comments added for that also

**Pull request merged**